code/modules/admin/IsBanned.dm

```diff
@@ -39,6 +39,17 @@ world/IsBanned(key,address,computer_id)
 			log_misc("Ban database connection failure. Key [ckey] not checked")
 			return
 
+		//Check if we got a gdpr ban for that person
+		var/DBQuery/bangdprquery = dbcon.NewQuery("SELECT id, created_at FROM ss13_ban_gdpr WHERE ckey_hashed = SHA2(:ckey:,256) OR address_hashed = SHA2(:address:,256) OR computerid_hashed = SHA2(:computerid:,256)")
+		bangdprquery.Execute(list("ckey"=ckey, "computerid"=computer_id, "address"=address))
+		while(bangdprquery.NextRow())
+			//They connected again, so we got their conset to use their data again. Let´s log their connection attempt
```

Comment:
There is no consent requirement for keeping security logs and banlists. It just needs to be proportional and needed for providing the service and you need to be open about how you proces said data.

I think the best scenario is working on making a clear article in the terms of service about processing of data (with preferable a seperate article about security data from banned people) first and then alter the code according to the agreement you write.

You change code before you have properly adressed the issue in your code of conduct, it should be the other way around.